emails/views.py

```diff
@@ -279,21 +287,27 @@ def _sns_message(message_json):
             status=400
         )
 
+    from_address = parseaddr(mail['commonHeaders']['from'][0])[1]
     to_address = parseaddr(mail['commonHeaders']['to'][0])[1]
-    local_portion = to_address.split('@')[0]
-
-    if local_portion == 'noreply':
-        incr_if_enabled('email_for_noreply_address', 1)
-        return HttpResponse('noreply address is not supported.')
-
-    domain_portion = to_address.split('@')[1]
+    [to_local_portion, to_domain_portion] = to_address.split('@')
+    # to_domain_portion = to_address.split('@')[1]
     try:
         # FIXME: this ambiguous return of either
         # RelayAddress or DomainAddress types makes the Rustacean in me throw
         # up a bit.
-        address = _get_address(to_address, local_portion, domain_portion)
+        address = _get_address(to_address, to_local_portion, to_domain_portion)
         user_profile = address.user.profile_set.first()
     except Exception:
+        if to_local_portion == 'replies':
+            user = User.objects.get(email=from_address)
+            profile = user.profile_set.first()
+            if not profile.has_premium:
+                # TODO: send the user an email that replies are a premium feature
+                return HttpResponse(
+                    "Rely replies require a premium account", status=403
+                )
+            return _handle_reply(message_json)
+
         return HttpResponse("Address does not exist", status=404)
 
     address_hash = sha256(to_address.encode('utf-8')).hexdigest()
```

Comment:
We could potential move the rest of the code snippet here to a separate function like ` _handle_incoming_email` to follow the `_handle_reply`. Wdyt?